src/Adapter/MSTest.CoreAdapter/Discovery/TypeEnumerator.cs

```diff
@@ -106,11 +106,19 @@ internal UnitTestElement GetTestFromMethod(MethodInfo method, ICollection<string
             var isAsync = ReflectHelper.MatchReturnType(method, typeof(Task));
 
             var testMethod = new TestMethod(method.Name, this.type.FullName, this.assemblyName, isAsync);
+
             if (!method.DeclaringType.FullName.Equals(this.type.FullName))
             {
                 testMethod.DeclaringClassFullName = method.DeclaringType.FullName;
             }
 
+            if (!this.reflectHelper.IsMethodDeclaredInSameAssemblyAsType(method, this.type))
```

Comment:
How about passing this in from the parent caller when it calls GetTestFromMethod? Wouldn't need to query for it again then.